loggers/fields.go

```diff
@@ -49,3 +49,15 @@ func FromContext(ctx context.Context) LogFields {
 	}
 	return nil
 }
+
+func GetNewLogContextWithValue(ctx context.Context, key string, value interface{}) context.Context {
+	logfields := FromContext(ctx)
+	newLogFields := make(LogFields)
+	for key, val := range logfields {
```

Comment:
the key name is reused  here, can we add a test to these functions to verify its not modifying the incoming key?

Reply:
done